cpp/src/arrow/util/bit_util.h

```diff
@@ -119,7 +119,11 @@ constexpr int64_t CeilDiv(int64_t value, int64_t divisor) {
   return (value == 0) ? 0 : 1 + (value - 1) / divisor;
 }
 
-constexpr int64_t BytesForBits(int64_t bits) { return (bits + 7) >> 3; }
+// Return the number of bytes needed to fit the given number of bits
+constexpr int64_t BytesForBits(int64_t bits) {
+  // This formula avoids integer overflow on very large `bits`
+  return (bits >> 3) + ((bits & 7) != 0);
```

Comment:
I think this is just the beginning.

Reply:
You mean of avoiding overflow in various places? I agree.

Reply:
yes.